module-system/sov-modules-api/src/cli.rs

```diff
@@ -0,0 +1,32 @@
+use crate::CliWallet;
+
+/// An argument to the cli containing a json string
+#[derive(clap::Args, PartialEq, core::fmt::Debug, Clone, PartialOrd, Ord, Eq, Hash)]
+pub struct JsonStringArg {
+    /// The json formatted transaction data
+    #[arg(long, help = "The JSON formatted transaction")]
+    pub json: String,
+}
+
+/// An argument to the cli containing a path to a file
+#[derive(clap::Args, PartialEq, core::fmt::Debug, Clone, PartialOrd, Ord, Eq, Hash)]
+pub struct FileNameArg {
+    /// The json formatted transaction data
+    #[arg(long, help = "The JSON formatted transaction")]
+    pub path: String,
+}
+
+impl TryFrom<FileNameArg> for JsonStringArg {
+    type Error = std::io::Error;
+    fn try_from(arg: FileNameArg) -> Result<Self, Self::Error> {
+        let json = std::fs::read_to_string(arg.path)?;
+        Ok(JsonStringArg { json })
+    }
+}
+
+pub trait CliFrontEnd<RT>
```

Comment:
The `RT` is not used anywhere. Do we need it?

Reply:
I don't think we need it *yet*, but we'll have to have it for the version that removes the extra generics. Otherwise, we'll start running into issues with specialization